nemo/collections/common/parts/preprocessing/manifest.py

```diff
@@ -75,9 +84,44 @@ def item_iter(
         logging.debug('Using manifest file: %s', str(manifest_file))
         cached_manifest_file = DataStoreObject(manifest_file).get()
         logging.debug('Cached at: %s', str(cached_manifest_file))
+
+        if shard_strategy is not None and world_size > 1:
+            if shard_strategy == 'scatter':
+                # Read only necessary manifest lines
+                logging.debug('Partitioning manifest: global rank = %d, world size = %d', global_rank, world_size)
+
+                # Get number of lines in the current manifest
+                num_lines = get_num_lines(cached_manifest_file)
```

Comment:
Rather import data_util and then explicitly call data_util.get_num_lines for clarity